project-generated-code-backend/Backend/Repository/DatabaseSql/FloorDatabaseSql.cs

```diff
@@ -16,6 +16,11 @@ public override List<Floor> GetAll()
                 .ToList();
         }
 
+        
+        public override Floor GetById(string serialNumber)
+        {
+            return GetAll().Where(f => f.SerialNumber.Equals(serialNumber)).ToList()[0];
```

Comment:
this could be just:
`return dbContext.Floor.Find(serialNumber);`
the method find works with primary key and gets exactly one object